hack/lib/log.sh

```
@@ -0,0 +1,216 @@
+#!/bin/bash
+#
+# This library holds all of the logging functions for OpenShift bash scripts.
+
+# os::log::install_cleanup installs os::log::clean_up_logger as a trap on interrupts, termination, and exit.
+# If any traps are currently set for these signals, os::log::clean_up_logger is prefixed.
+#
+# Globals:
+#  None
+# Arguments:
+#  None
+# Returns:
+#  None
+function os::log::install_cleanup() {
+    trap "os::log::clean_up_logger; $(trap -p SIGINT | awk -F"'" '{print $2}')" SIGINT
+    trap "os::log::clean_up_logger; $(trap -p SIGTERM | awk -F"'" '{print $2}')" SIGTERM
+    trap "os::log::clean_up_logger; $(trap -p EXIT | awk -F"'" '{print $2}')" EXIT
+}
+
+# os::log::clean_up_logger should be trapped so that it can stop the logging utility once the script that
+# installed it is finished.
+# This function stops logging and generates plots of data for easy consumption.
+#
+# Globals:
+#  - LOG_DIR
+#  - LOGGER_PID
+#  - SAR_LOGFILE
+# Arguments:
+#  None
+# Returns:  
+#  None
+function os::log::clean_up_logger() {
+    local return_code
```

call the directive to ignore errors.